### PR TITLE
Putting logic for pex behind a platform check

### PR DESCRIPTION
**Pull request by Nairu**

### CODE REVIEW

File: src/kraken/wrapper/_buildenv.py [hunk 1]
```diff
@@ -116,91 +114,97 @@ def set(self, metadata: BuildEnvMetadata) -> None:
         self._metadata = metadata
 
 
-class PexBuildEnv(BuildEnv):
-    STYLES = (EnvironmentType.PEX_ZIPAPP, EnvironmentType.PEX_PACKED, EnvironmentType.PEX_LOOSE)
+if platform.system().lower() != "windows":
```
Comment (NiklasRosenstein) [inline src/kraken/wrapper/_buildenv.py:117]: Could you move this into a `_buildenv_pex.py` instead and conditionally import from it?

Comment (Nairu) [reply src/kraken/wrapper/_buildenv.py]: Extracted the `_buildenv.py` logic out to a couple of files to make it slightly neater for the windows check

File: src/kraken/wrapper/_buildenv_manager.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import logging
+import os
+from pathlib import Path
+from urllib.parse import quote, urlparse, urlunparse
+
+from kraken.common import EnvironmentType, RequirementSpec, not_none, safe_rmpath
+
+from ._buildend_venv import VenvBuildEnv
+from ._buildenv import BuildEnv, BuildEnvMetadata, BuildEnvMetadataStore
+from ._buildenv_pex import _pex_build_env_supported
```
Comment (NiklasRosenstein) [inline src/kraken/wrapper/_buildenv_manager.py]: This will still cause that it tries to import `fcntl` on Windows 😀

Comment (NiklasRosenstein) [reply src/kraken/wrapper/_buildenv_manager.py]: Addressed in f04a827

**Pull request merged**